ui/src/overview.tsx

```diff
@@ -359,83 +429,132 @@ function DemographicCharts({ cohort }: DemographicChartsProps) {
-      const genderAgeRange = `${gender} ${ageRange}`;
-
-      if (!demographicsByGenderAgeRace.has(genderAgeRange)) {
-        demographicsByGenderAgeRace.set(genderAgeRange, new Map());
-      }
-
-      if (!demographicsByGenderAgeRace.get(genderAgeRange).has(race)) {
-        demographicsByGenderAgeRace.get(genderAgeRange).set(race, 0);
-      }
-
-      const currCountForRace = demographicsByGenderAgeRace
-        .get(genderAgeRange)
-        .get(race);
-
-      demographicsByGenderAgeRace
-        .get(genderAgeRange)
-        .set(race, currCountForRace + count);
+      chartConfigs.forEach((config, index) => {
+        // Determine the primary data groupings for the chart.
+        const primaryPropertyComponents: string[] = [];
+        config.primaryProperties.forEach((property) => {
+          // Temporarily check if property is age since API only supports year of birth.
+          const propertyName =
+            property.key === "age" ? "year_of_birth" : property.key;
+          const propertyValues = Object.entries(
+            demographicData[i]?.definition ?? {}
+          ).find((data) => data[0] === propertyName);
+
+          if (propertyValues) {
+            const dataIntegerVal = propertyValues[1].int64Val;
+            const dataStringVal = propertyValues[1].stringVal;
+            const dataBoolVal = propertyValues[1].boolVal;
+            // The UI currently assumes that property.value represents an age bucket if defined.
+            if (property.value) {
+              if (dataIntegerVal) {
+                // Temporarily calculating ages while the backend provides years of birth.
+                const currentYear = new Date().getFullYear();
+                const yob = dataIntegerVal;
+                const age = parseInt(currentYear.toString()) - yob;
+                property.value.forEach((range) => {
+                  const min = range.min;
+                  const max = range.max;
+                  if (min && max && min <= age && age <= max) {
+                    primaryPropertyComponents.push(`${min}-${max}`);
+                  } else if (min && !max && min <= age) {
+                    primaryPropertyComponents.push(`>${min}`);
+                  } else if (!min && max && age <= max) {
+                    primaryPropertyComponents.push(`<${max}`);
+                  }
+                });
+              }
+            } else if (dataIntegerVal) {
+              primaryPropertyComponents.push(`${dataIntegerVal}`);
+            } else if (dataStringVal) {
+              primaryPropertyComponents.push(dataStringVal);
+            } else if (dataBoolVal) {
+              primaryPropertyComponents.push(`${dataBoolVal}`);
+            }
+          }
+        });
+
+        // Add the primary property grouping to the chart data if it hasn't already
+        const primaryPropertyString = primaryPropertyComponents.join(" ");
+        if (!chartsData[index].has(primaryPropertyString)) {
```

Comment:
This code would be cleaner if it looped over charts -> demographics rather than demographics -> charts like it does now. You wouldn't need checks like this because you would only be building and pushing the data for each chart once.